v2/src/log_stacktrace.c

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#include "windows.h"
+#include "winnt.h"
+#include "dbghelp.h"
+
+#include "c_logging/logging_stacktrace.h"
+
+static volatile LONG doSymInit = 0; /*0 = not initialized, 1 = initializing, 2= initialized*/
+
+typedef union SYMBOL_INFO_EXTENDED_TAG
+{
+    SYMBOL_INFO symbol;
+    unsigned char extendsUnion[sizeof(SYMBOL_INFO) + MAX_SYM_NAME - 1]; /*this field only exists to extend the size of the union to encompass "CHAR    Name[1];" of the SYMBOL_INFO to be as big as TRACE_MAX_SYMBOL_SIZE - 1 */ /*and the reason why it is not malloc'd is to exactly avoid a malloc that cannot be LogError'd (how does one log errors in a log function?!)*/
+}SYMBOL_INFO_EXTENDED;
+
+static SRWLOCK lockOverSymCalls = SRWLOCK_INIT;
+
+/*returns number of characters copied into destination*/
+static size_t memcat(char* destination, size_t destinationSize, const char* source, size_t sourceSize)
+{
+    size_t sizeToCpy = destinationSize<sourceSize?destinationSize:sourceSize;
+    (void)memcpy(destination, source, sizeToCpy);
+    return sizeToCpy;
+}
```

Comment:
Well, this is used in one place, so I could say that maybe we don't need a function and we just memcpy.

Reply:
removed and did it inline